### PR TITLE
Update agent pool and add tasks to security check

### PR DESCRIPTION
**Pull request by dingmeng-xue**
## Description

Update agent pool setting and use private agent pool
Add and upgrade tasks in security check

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by dingmeng-xue** [comment]
\azp run azure-powershell - security-tools

**Review by YanaXu — APPROVED** [review]
Good to me.

File: .azure-pipelines/powershell-core.yml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+  MacOSAgentPoolVMImage: macOS-11
   TestFramework: netcoreapp3.1
   TestTarget: Test
   Configuration: Debug
-  DebugLocalBuildTasks: true
-  IsGenerateBased: $[eq(variables['system.pullRequest.targetBranch'], 'generation')]
   BuildTimeoutInMinutes: 120
   AnalysisTimeoutInMinutes: 120
   TestTimeoutInMinutes: 180
-  BuildAzPredictor: false
```
Comment (BethanyZhou) [inline .azure-pipelines/powershell-core.yml:16]: Are Lin11-12 and 16 removed intentionally?

Comment (BethanyZhou) [reply .azure-pipelines/powershell-core.yml:16]: - I remember the build logics are different between generation branch and other branches. They are distinguished by  IsGenerateBased. 
- BuildAzPredictor is used to skip AzPredictor build task. 
@wyunchi-ms  Could you help confirm these logics if still exist?

Comment (wyunchi-ms) [reply .azure-pipelines/powershell-core.yml:16]: This variable is still in use. Please do not remove it.

File: .azure-pipelines/powershell-core.yml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+  LinuxAgentPoolVMImage: ''
   MacOSName: macOS
-  MacOSImage: macOS-11
+  MacOSAgentPoolName: 'Azure Pipelines'
+  MacOSAgentPoolVMImage: macOS-11
   TestFramework: netcoreapp3.1
   TestTarget: Test
   Configuration: Debug
-  DebugLocalBuildTasks: true
-  IsGenerateBased: $[eq(variables['system.pullRequest.targetBranch'], 'generation')]
```
Comment (wyunchi-ms) [inline .azure-pipelines/powershell-core.yml:12]: These variable are also needed.

File: .azure-pipelines/security-tools.yml [hunk 3]
```diff
@@ -53,14 +53,22 @@ jobs:
       outputFormat: sarif
       scanFolder: SecurityTmp
       suppressionsFile: tools/SecurityTools/CredScanSuppressions.json
-      debugMode: false
-      folderSuppression: false
```
Comment (BethanyZhou) [inline .azure-pipelines/security-tools.yml:57]: Should we keep these two parameters? @dingmeng-xue

Comment (dingmeng-xue) [reply .azure-pipelines/security-tools.yml:57]: No. they are default configuration or not supported.

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Pull request merged**